Updated Get-AzAutomationJobOutputRecord to handle JSON and Text record values

lwajswaj opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

This is a re-open of [PR 8607](https://github.com/Azure/azure-powershell/pull/8607). 

Answering @NarayanThiru comments: Current logic doesn't work as the error message and code is different (regardless that it's easier and cleaner to check on the output format rather than attempt and failing). Regarding the issue and how to repro/test, refer to the linked issues they contain a lot of good information.

CC: @vladimir-shcherbakov, @MiYanni 

## Description

This PR addresses issue #7977 and #8600. Root cause of the issue was identified as stream outputs always expecting to be a JSON file; however, this ain't true in all cases as String or PSObjects are not returned as JSON. To fix it, a RegEx was put in place to identify JSON output from text output. 

## Checklist

- [X] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [X] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

markcowl commented:
@vrdmr @jemex Can one of you review this PR?

NarayanThiru commented:
The changes look good.  @thomasyip-msft, @vrdmr, @safeermohammed, can any of you merge this PR?

markcowl commented on src/Automation/Automation/ChangeLog.md:
```diff
@@ -31,6 +31,9 @@
 
 ## Version 1.1.2
 * Fixed issue when retreiving certain monthly schedules in several Azure Automation cmdlets
+* Updated Get-AzAutomationJobOutputRecord to handle JSON and Text record values.
```
Can we move this change under the 'Upcoming Release' heading above?  Otherwise, LGTM

lwajswaj replied:
Just moved to the right spot, not sure how I got it there in the first place.

markcowl commented:
@lwajswaj if you can make the changelog change, we can fix this.  Is there a simple way we can add test automation for this fix, to prevent future regressions?

lwajswaj commented:
> 
> 
> @lwajswaj if you can make the changelog change, we can fix this. Is there a simple way we can add test automation for this fix, to prevent future regressions?

I might be able to add a quick test under the scenario tests. Kind of like starting the mock up runbook and retrieving its output stream (that's the current error condition)

lwajswaj commented:
> 
> 
> @lwajswaj if you can make the changelog change, we can fix this. Is there a simple way we can add test automation for this fix, to prevent future regressions?

@markcowl, Just added tests for this scenario...... along the way, I found the tests for runbook execution were not executing properly, for example expected result was never evaluated, that's why I included a couple of extra changes:

* Renamed **fastjob.ps1** to **Test-PowershellRunbook.ps1** (name is meaningful now) also migrated this script from using workflow to plain powershell. Added additional outputs since that's the error condition for  'Get-AzAutomationJobOutputRecord' otherwise test was successful with a single integer output.
* Renamed **fastjob.py** to **TestPythonRunbook.py** (name is meaningful too). Also added additional outputs (same reason as before)
* I found 'Test-RunbookWithParameter' was not using its parameter $expectedResult, I changed it to include this assertion as part of the test

Of course, added the tests for 'Get-AzAutomationJobOutputRecord' for preventing regressions.

markcowl commented:
/azp run
